web/src/games/kriegspiel/Board.tsx

```diff
@@ -0,0 +1,770 @@
+      },
+      onPinch: (state) => {
+
+        const newScale = state.offset[0]
+        //if (newScale > 0.5 && newScale < 10) {  }
+
+        setMapScale(newScale);
+      },
+
+    },
+    {
+      target: boardRef,
+      eventOptions: { passive: false },
+      preventDefault: true,
+    }
+  ); */
+
+  //render Main board
+
+  const gameBoard = (
+    <svg
+      viewBox={`-0.6 -0.6 ${BoardSize.mx + 1.2} ${BoardSize.my + 1.2}`}
+      ref={boardRef}
+      //onWheel={myOnWheel} onMouseMove={drag} onMouseDown={startDarg} onMouseUp={endDarg} onMouseLeave={endDarg}
+      //onTouchStart={startDarg} onTouchEnd={endDarg} onTouchCancel={endDarg}
+    >
+      <g
+      //transform={`translate(${BoardSize.mx / 2} ${BoardSize.my / 2})  scale(${mapScale}) translate(${mapPos.x - BoardSize.mx / 2} ${mapPos.y - BoardSize.my / 2})`}
+      >
+        <rect
+          x={-0.6}
+          y={-0.6}
+          width="100%"
+          height="100%"
+          fill={pico8Palette.light_peach}
+          stroke={pico8Palette.dark_grey}
+          strokeWidth="0.05"
+        />
+        {/* background */}
+        {Array(BoardSize.mx)
+          .fill(null)
+          .map((_, id) => gTranslate(renderStr((id + 1).toString()), id, -0.8))}
+        {Array(BoardSize.my)
+          .fill(null)
+          .map((_, id) => gTranslate(renderStr(String.fromCharCode(65 + id)), -0.8, id))}
+        {Array(BoardSize.mx)
+          .fill(null)
+          .map((_, id) => gTranslate(renderStr((id + 1).toString()), id, BoardSize.my - 1 + 0.8))}
+        {Array(BoardSize.my)
+          .fill(null)
+          .map((_, id) => gTranslate(renderStr(String.fromCharCode(65 + id)), BoardSize.mx - 1 + 0.8, id))}
+        {renderLayer((_, id) => (
+          <rect
+            key={id}
+            width="1"
+            height="1"
+            fill={getCellColor(id)}
+            stroke={pico8Palette.dark_grey}
+            strokeWidth="0.05"
+          />
+        ))}
+        <line
+          x1="0"
+          y1={BoardSize.my / 2}
+          x2={BoardSize.mx}
+          y2={BoardSize.my / 2}
+          stroke={pico8Palette.lavender}
+          strokeWidth="0.2"
+        />
+        {/* supply line */}
+
+        {getDirSuppliedLines(G, '0')[1].map((lines) =>
+          lines.map((lineLst) => {
+            /*  let stPos = CId2Pos(lineLst[0]);
+           let edPos = CId2Pos(lineLst[lineLst.length - 1]);
+           return stPos && edPos && gTranslate(<line x1={stPos.x} y1={stPos.y} x2={edPos.x} y2={edPos.y} stroke={fictionColor('0')} stroke-width="0.1" stroke-dasharray="0.5 0.1" />, 0.45, 0.45)
+          */
+            return (
+              lines.length > 1 &&
+              gTranslate(
+                drawLine(lineLst[0], lineLst[lineLst.length - 1], fictionColor('0'), 0.05, '0.5, 0.1'),
+                -0.05,
+                -0.05,
+              )
+            );
+          }),
+        )}
+        {getDirSuppliedLines(G, '1')[1].map((lines) =>
```

Comment:
This function is too large and too hard to read. Please break it down into smaller React components. Each .map() function here is a good candidate for a separate component.